src/ScottPlot5/ScottPlot/Range.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ScottPlot
+{
+    public struct Range
```

Comment:
Let's add XML docs indicating what this is. It's not a pixel range or a coordinate rage ... it's any arbitrary range. Maybe something like

```cs
///<summary>
///Represents a range between any two finite values (inclusive)
///</summary>
```

Should we force `min` to be < `max` in the constructor?

Should we throw if the user passes in NaN or infinity?

Also let's add XML docs to functions. I'll help:

### Normalize

```cs
///<summary>
///Returns the given value as a fraction of the range between min and max.
///Returns values <0 or >1 if the given value is outside the range.
///</summary>
```

### Clamp

```cs
///<summary>
///Returns the given value clamped to the range (inclusive).
///Attempting to clamp NaN will return NaN.
///</summary>
```

Rather than `NormalizeAndClampToUnitRange()`, maybe Normalize should just have a `bool clamp`?

Reply:
Clamping NaN to NaN is not the default behaviour of a naive implementation, and I don't know if we want to take a stance on what should happen when you clamp NaN? There is currently no codepath that (intentionally) attempts to clamp NaN and so I think it's fine for the behaviour to be "did you really think that was going to work?"

Reply:
Also, requiring $min \neq max$ does prevent a possible divide by zero, but I think it's going to be inconvenient in the future.

For example, if someone passes in a heatmap with a single element $x$, when we calculate the extent of it we're going to get the degenerate interval $[x, x]$. I'd rather try to make sure that `Range` behaves properly on degenerate intervals than walk on eggshells around it everywhere trying to avoid such intervals.

Reply:
These are good points! There's a lot flying around right now. I'm going to add a TODO so we can revisit this topic some day in the future.

Same with the coordinate vs. vector addition thing.

I'm just trying to get off the ground, but we can define these things carefully later and add tests to carefully lock-in all these behavior decisions. (cringy to write that - code now and test later is such a trope - but I don't see SP5 getting off the ground if we jump into all these little minutia corner cases this early in the game)